### ISSUE TITLE
Any reason toKeyPair on blsKey dont return an full ID?

### ISSUE THREAD

**Issue by vongohren** [transmute-industries/did-key.js#103]
https://github.com/transmute-industries/did-key.js/blob/master/packages/bls12381/src/Bls12381G2KeyPair.ts#L94

Im using this lib to generate a key to be used towards mattrs json-ld bbs signature.

We ended up having to do this little magic: https://gist.github.com/S3bb1/ddd74704c0f9980d93e4e6302439594c#file-index-js-L54 to get the JSON LD to work.

Or is it something we are missing that we do wrong?

In this gist we convert the buffer to bs58, but toKeyPair does that for us. So using that method in the updated version. But I still have to update the ID for things to work. Could not toKeyPair just return a different ID, or where are things going wrong in the loop.

I have another issue going over at mattr: https://github.com/mattrglobal/jsonld-signatures-bbs/issues/122

**Comment by OR13** [comment]
@vongohren yea... this is related to the relative refs, debate... see 

https://github.com/w3c-ccg/did-method-key/pull/36#issuecomment-855014917

TL;DR ... relative ref ids are terrible.... for exactly the reason you have indicated...

we have similar code all over the place... and we are convinced that relative refs are terrible, especially if you are doing linked data operations often (which we are).


The next versions of all our key pair classes and did key will use absolute IRIs everywhere.

**Comment by vongohren** [comment]
@OR13 thanks for pointing this out. Then atleast im not the crazy one. We will see where this goes then!
